vowpalwabbit/io_buf.h

```diff
@@ -55,20 +55,22 @@ class io_buf
   bool _verify_hash;
   uint32_t _hash;
   static constexpr size_t INITIAL_BUFF_SIZE = 1 << 16;
-
- public:
+  char* head;
   v_array<char> space;  // space.begin = beginning of loaded values.  space.end = end of read or written values from/to
                         // the buffer.
+ public:
   v_array<int> files;
   size_t count;    // maximum number of file descriptors.
   size_t current;  // file descriptor currently being used.
-  char* head;
-  v_array<char> currentname;
-  v_array<char> finalname;
 
   static constexpr int READ = 1;
   static constexpr int WRITE = 2;
 
+  size_t available_to_read() const
+  {
+    return head - space.begin();
```

Comment:
This is the starting index of the first available byte to read, not the number of bytes available to read

Reply:
Good point, the naming is wrong, this method is used to tell how many written bytes are in the buffer.  All written data is kept between `being()` and `head`.

Reply:
Renamed to unflushed_bytes_count.